### PR TITLE
Create operation queue with disk storage

### PR DESCRIPTION
**Pull request by aniezurawski**

### CODE REVIEW

File: neptune/internal/collections/disk_queue.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        try:
+            os.mkdir(self._dir_path)
+        except FileExistsError:
+            pass
+
+        self._read_file_idx = 0
+        self._write_file_idx = 0
+        self._writer = open(self._current_write_log_file(), "a")
+        self._reader = JsonFileSplitter(self._current_read_log_file())
```
Comment (PiotrJander) [inline neptune/internal/collections/disk_queue.py:50]: I would consider making this a function which always returns the current file (no need to manually maintain the invariant). Admittedly then we couldn't close files manually, but files get closed by GC anyway.

Same for self._writer

File: neptune/internal/collections/disk_queue.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import os
+from typing import TypeVar, List, Callable, Optional
+
+from neptune.internal.collections.queue import Queue
+from neptune.internal.utils.json_file_splitter import JsonFileSplitter
+
+T = TypeVar('T')
+
+
+class DiskQueue(Queue[T]):
```
Comment (PiotrJander) [inline neptune/internal/collections/disk_queue.py:26]: Consider exposing two interfaces, one for writing and one for reading. No strong feelings about this one though.

File: neptune/internal/utils/json_file_splitter.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#
+import json
+from json import JSONDecodeError
+from typing import Optional
+
+
+class JsonFileSplitter:
+
+    # TODO: experiment with larger buffer sizes
+    BUFFER_SIZE = 4096
```
Comment (PiotrJander) [inline neptune/internal/utils/json_file_splitter.py:24]: Out of curiosity: why this size?

File: neptune/internal/utils/json_file_splitter.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    def _read_data(self):
+        data = self._file.read(self.BUFFER_SIZE)
+        if not data:
+            return
+        self._buffer = self._buffer[self._start_pos:] + data
+        self._start_pos = 0
+
+    def _decode(self):
+        self._start_pos = self._buffer.index("{", self._start_pos)
+        data, size = self._decoder.raw_decode(self._buffer, self._start_pos)
```
Comment (PiotrJander) [inline neptune/internal/utils/json_file_splitter.py:54]: The name `size` is misleading, how about `end`?

**Review by PiotrJander — APPROVED** [review]

File: neptune/internal/utils/json_file_splitter.py (hunk 4 above)
Comment (aniezurawski) [reply neptune/internal/utils/json_file_splitter.py:54]: Ok.

File: neptune/internal/utils/json_file_splitter.py (hunk 3 above)
Comment (aniezurawski) [reply neptune/internal/utils/json_file_splitter.py:24]: It is default internal buffer size in splitstream library (https://github.com/rickardp/splitstream) which eventually I dicided not to use. But 4kb seems to be good guess.

**Pull request merged**